### PR TITLE
EVG-7037 add restricted project vars for safe commands

### PR DESCRIPTION
**Pull request by ablack12**
For restricted variables, I don't want to duplicate the logic of private variables as well/allow for programmer error with ensuring restricted variables are also private variables -- I think it's okay to trust the user to make restricted variables private.

### CODE REVIEW

File: service/templates/projects.html [hunk 1]
```diff
@@ -340,6 +340,14 @@ <h2> Settings for [[displayName]]</h2>
                <input type="checkbox" name="set_private" ng-model="proj_var.is_private" ng-disabled="!validKeyValue(proj_var.name, proj_var.value)" /> <strong>Private Variable</strong>
              </label>
             </div>
+            <div>
+              <label class="control-label">
+                <input type="checkbox" name="set_restricted" ng-model="proj_var.is_restricted" ng-disabled="!validKeyValue(proj_var.name, proj_var.value)" /> <strong>Restricted Variable</strong>
+              </label>
+            </div>
+            <span class="warning-text" ng-show="proj_var.is_restricted"><i class="fa fa-warning"></i>
+              Restricted variables are only used with commands that won't leak the variables. Should also be set to private.
```
Comment (bsamek) [inline service/templates/projects.html:349]: I'm not sure they should also be set to private. They are only accessible to project admins, so it's conceivable that project admins are fine being able to read them from the UI.

File: command/s3_put.go [hunk 2]
```diff
@@ -168,11 +168,11 @@ func (s3pc *s3put) validate() error {
...
 }
 
-// Apply the expansions from the relevant task config to all appropriate
-// fields of the s3put.
+// Apply the expansions from the relevant task config (including restricted expansion)
+// to all appropriate fields of the s3put.
 func (s3pc *s3put) expandParams(conf *model.TaskConfig) error {
 	var err error
-	if err = util.ExpandValues(s3pc, conf.Expansions); err != nil {
+	if err = util.ExpandValues(s3pc, conf.GetAllExpansions()); err != nil {
```
Comment (bsamek) [inline command/s3_put.go:175]: Could you call this `conf.GetRestrictedExpansions` instead? "All" is technically correct, but I can imagine a developer looking at what functions are available on `conf` and not realizing what "All" means in this context.

File: service/templates/projects.html (hunk 1 above)
Comment (ablack12) [reply service/templates/projects.html:349]: Okay I can change this warning to clarify they won't be private. I do want to make sure it's understood that Restricted is not a subset of private.

Comment (ablack12) [reply service/templates/projects.html:349]: Or I can clarify what "restricted" means (and will add to wiki), users will see I suppose that it's not redacted when they press the plus button.

**Pull request merged**